codestodians/food-basket-front-end
# [Feature: #154657644] Enable page redirect after authentication from dedicated loginpage

#### What does this PR do?
- Enables redirect to the homepage once authentication from the dedicated login-page is successful.

#### Description of Task to be completed?
- Introduction of `Router` in place of `BrowserRouter` from `react-router-dom`.
- Updating of the import location of the `App` component in the application entry file.
- Creation of an history object to hold the history of visited paths in the application.
- Addition of logic to handle redirect from the `/login` path.

#### How should this be manually tested?
- Checkout into the `ft-enable-page-redirect-after-authentication-154657644` branch.
- Update your packages with `yarn` or `yarn install`
- Run `yarn start`
- Navigate to `localhost:8080/login`
- Signup or Login an existing user.
- Observe the redirect after successful authentication

#### Any background context you want to provide?
- The `Router` component from `react-router-dom` provides a more straightforward way for handling route change and redirects than the `BrowserRouter` component, and so I thought it'd be a good option to explore this provision.
- It was also necessary to set the foundation for being able to track the last visited route by simply using the back button on the browser. This is a feature that the `history` prop and object provide out-of-the-box. Hence, this PR.

#### What are the relevant pivotal tracker stories?
- [#154657644](https://www.pivotaltracker.com/story/show/154657644)
